mdal/frmts/mdal_ply.cpp

```diff
@@ -0,0 +1,445 @@
+    else if ( startsWith( line, "end_header" ) )
+    {
+      break;
+    }
+    //
+    //if binary - give up
+    //
+    else if ( startsWith( line, "binary" ) )
+    {
+      MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " only ASCII format PLY files are supported" );
+      return nullptr;
+    }
+
+    //
+    // if "comment crs" assume that the rest is the crs data
+    //
+    else if ( startsWith( line, "comment crs " ) )
+    {
+      line.erase( 0, 12 );
+      proj = line;
+      if ( !std::getline( in, line ) )
+      {
+        MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " Header is corrupt" );
+        return nullptr;
+      }
+    }
+
+    //
+    // probably a comment line
+    //
+    else
+    {
+      if ( !std::getline( in, line ) )
+      {
+        MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " Header is corrupt" );
+        return nullptr;
+      }
+    }
+  }
+  while ( true );
+
+  Vertices vertices( 0 );
+  Faces faces( 0 );
+  Edges edges( 0 );
+  size_t maxSizeFace = 0;
+  size_t faceSize = 0;
+
+  //datastructures that will contain all of the datasets, categorised by vertex, face and edge datasets
+  std::vector<std::vector<double>*> vertexDatasets; // conatins the data
+  std::vector<std::string> vProp2Ds; // contains the dataset names
+  std::vector<std::vector<double>*> faceDatasets;
+  std::vector<std::string> fProp2Ds;
+  std::vector<std::vector<double>*> edgeDatasets;
+  std::vector<std::string> eProp2Ds;
+
+
+
+  //
+  // load the elements in order
+  //
+  if ( !std::getline( in, line ) )
+  {
+    MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " does not contain any definitions" );
+    return nullptr;
+  }
+  chunks = split( line, ' ' );
+
+  //
+  // configure the vectors to hold the data
+  //
+  for ( size_t elid = 0; elid < elements.size(); ++elid )
+  {
+    MDAL::DriverPly::element element = elements[elid];
+    if ( element.name == "vertex" )
+    {
+      vertexCount = element.size;
+      vertices.resize( vertexCount );
+      for ( size_t i = 0; i < element.properties.size(); ++i )
+      {
+        if ( element.properties[i] != "x" &&
+             element.properties[i] != "y" &&
+             element.properties[i] != "z" &&
+             !element.list[i] )
+        {
+          vertexDatasets.push_back( new std::vector<double> );
+          vProp2Ds.push_back( element.properties[i] );
+        }
+      }
+    }
+    else if ( element.name == "face" )
+    {
+      faceCount = element.size;
+      faces.resize( faceCount );
+      for ( size_t i = 0; i < element.properties.size(); ++i )
+      {
+        if ( element.properties[i] != "vertex_indices" &&
+             !element.list[i] )
+        {
+          faceDatasets.push_back( new std::vector<double> );
+          fProp2Ds.push_back( element.properties[i] );
+        }
+      }
+    }
+    else if ( element.name == "edge" )
```

Comment:
I am not sure MDAL can support, for now, having faces and edges together. Edge is, for now, used for 1D Mesh, that is network of node linked only by edges.

Reply:
I have three comments to make :

1) The PLY format and this driver are very flexible about the data contents. This is not to say that there MUST be both edges and faces. Just that neither the format nor the driver enforce that there is NOT both edges and faces, 

2) As the tests show, MDAL can support data that has both faces and edges. In the test data I am loading and then successfully reading datasets that contain data on both the faces and edges. That data has no particular GIS meaning , but MDAL processes it correct;y as I understand that it should. It is up to the host program if that data makes sense, and

3) I think that you are saying that it is possible to load data from a PLY file that does not make sense to other formats and cannot be converted to other formats successfully. Isn't that in the nature of a DAL? You either end up enforcing a lowest common denominator type of entity model and have no effective value or you accept that the host application and the user needs to apply to intelligence about what data can convert to what.

Reply:
You are right @runette . What Vincent tried to express that the "host" program QGIS now, and other formats have either 1D or 2D data (so for example we read it as 2 separate meshes usually, one for 1D data). Such combined mesh is fine, but there could be some bugs in QGIS displaying that since noone tried it yet.

Best to check in QGIS to see if it is supported and possibly report bugs

Reply:
Probably easier and more effective to just put a note in the driver description. I cannot think of a real use-case that uses both - certainly for QGIS data although there may be for what I am using MDAL for. If a use-case for data both on faces and edges in QGIS comes up then the use-case owner can raise that with QGIS.

Reply:
Done in the next commit